rc4airconbotmvp.py

```diff
@@ -111,42 +97,38 @@ def start(update, context):
 def prompt_id(update, context):
     chat_id = update.message.chat.id
     user_input = update.message.text.replace(" ", "")
-
-    if (user_input == "/cancel"):
-        update.message.reply_text("Your session has been terminated. Please type /start to begin a new one.", reply_markup=ReplyKeyboardRemove())
-        return ConversationHandler.END
-
-    if (len(user_input) != 7 and len(user_input) != 6) or (user_input[0] != '#') or (user_input[3] != '-'):
+    
+    def error():
         context.bot.send_message(
         chat_id=chat_id,
-        text='Please key in using this format #xx-xx or #xx-xxx.'
+        text='Please key in using this format #xx-xx or #xx-xxx, or type /cancel to end your session.'
         )
 
+    if (user_input == "/cancel"):
+        cancel(update)
+        return 
+
```

Comment:
Please use regex instead to check the input.

Try to figure out the correct regex for dd-dd or dd-ddx (where d is number and x is alphabet). If you give up, can refer below.

<details>
  <summary>Spoiler warning</summary>
  
  ```
  /^[0-9]{2}-[0-9]{2}[A-F]?$/
  ```
  
</details>